### ISSUE TITLE
Possible to improve the view of scans?

### ISSUE THREAD

**Issue by kochmich** [EDDiscovery/EDDiscovery#2748]
808/5000
I have no problem, you are doing a great job! I am in the role of explorer and researcher.
Therefore I have a wish or suggestion for you: Please improve the scan view. You could present interesting information at a glance by directly appearing on the planets. You have collected this data automatically, but you could visualize it directly in the view with a letter:
- Is it possible to write down the G-value on all planets?
- What is the total of planetary and star masses?
- How old is the star system?
- How many belt clusters does a belt have?
- What type of star is it?
- What kind of gas giant is it?
- Does the planet have signals? And if so, which and how many (bio / geo)?
- Is the planet in the habitable zone?

The next step would be to show a Orrery  ;-) [but why not?]

![easy example](https://user-images.githubusercontent.com/44752904/76698630-b5906e80-66a5-11ea-9543-7a8b043f9dc2.jpg)

**Comment by robbyxp1** [comment]
We can add more display options, it just gets more visually dense doing it.

**Comment by klightspeed** [comment]
> Is it possible to write down the G-value on all planets?

i.e. Surface Gravity, even on non-landable planets?

> What is the total of planetary and star masses?

What units - stellar masses, earth masses, yottatonnes?  The stars are generally going to be 99.9% the total mass of the system (except in the rare case where you have a heavy gas giant orbiting a very light Y type star.

> How old is the star system?

i.e. age from the primary star? (all stars in the system will generally have the same age)

> How many belt clusters does a belt have?

Would count the number of child belt cluster objects under the belt object

> What type of star is it?

From StarType, Subclass, and Luminosity?

> What kind of gas giant is it?

From abbreviated PlanetClass?

> Does the planet have signals? And if so, which and how many (bio / geo)?

i.e. from the `SAASignalsFound` event when you map a body?  I don't think EDSM currently exposes the counts in its bodies API from where commanders have mapped bodies.

> Is the planet in the habitable zone?

Basically checking that Periapsis = SMA * (1 - e) and Apoapsis = SMA * (1 + e) fall within the calculated habitable zone for the star?

**Comment by kochmich** [comment]
Thank you for the very quick feedback!

I thought about more display options - so the user can decide how much information he can see.

For your questions:

**Is it possible to write down the G-value on all planets?**
> i.e. Surface Gravity, even on non-landable planets?

Yes - So it would be possible to estimate the mass / size of a planet. There are "planet hunters" in the community who are looking for particularly massive bodies. 
You are of course right: the surface gravity of a body depends on its rotation and density.


**What is the total of planetary and star masses?**
> What units - stellar masses, earth masses, yottatonnes? The stars are generally going to be 99.9% the total mass of the system (except in the rare case where you have a heavy gas giant orbiting a very light Y type star.

For Stars: Sun-Masses; For Planets: Earth-Masses. 
These mass values would have to be differentiated. I appreciate seeing how much matter the Stellar Forge Engine has made available for a system. In addition, the individual mass value of a body compared to its surface gravity would give a new way of assessing it. For example: very heavy body but very low surface gravity due to turbo rotation.


**How many belt clusters does a belt have?**
> Would count the number of child belt cluster objects under the belt object

Sometimes an asteroid belt has 12 clusters or only 5 or so. A number above the asteroid belt would help to understand how "significant" the belt is.


**What type of star is it?**
> From StarType, Subclass, and Luminosity?

Yes, all :-) -  that would be fine
In principle for every body: Rocky, Icy, Gas Giant, with water based Life and so on ...
This could be supplemented with information about its path deviation or its speed.

**Does the planet have signals? And if so, which and how many (bio / geo)?**
> i.e. from the SAASignalsFound event when you map a body? I don't think EDSM currently exposes the counts in its bodies API from where commanders have mapped bodies.

I don't know exactly where the data for counting the signals comes from. The game seems to generate them when scanning the system. It looks like bio signals are "safe" and geo signals are "likely".


**Is the planet in the habitable zone?**
> Basically checking that Periapsis = SMA * (1 - e) and Apoapsis = SMA * (1 + e) fall within the calculated habitable zone for the star?

Well, I was thinking of a simple mark on the body that says: It's in the habitable zone.

**Comment by klightspeed** [comment]
>>> What is the total of planetary and star masses?
>>
>> What units - stellar masses, earth masses, yottatonnes? The stars are generally going to be 99.9% the total mass of the system (except in the rare case where you have a heavy gas giant orbiting a very light Y type star.
>
> For Stars: Sun-Masses; For Planets: Earth-Masses.
> These mass values would have to be differentiated. I appreciate seeing how much matter the Stellar Forge Engine has made available for a system. In addition, the individual mass value of a body compared to its surface gravity would give a new way of assessing it. For example: very heavy body but very low surface gravity due to turbo rotation.

Rotational period doesn't affect surface gravity in Elite Dangerous.  The only two factors affecting surface gravity are mass and radius.  Also, if a body were rotating so fast that the equatorial surface gravity were significantly affected, then it would be very oblate.

Frontier has slowed down a number of planets that had rotational periods less than about 1.4x what the orbital period would be at the surface - though there are still some gas giants with a rotation period less than 1.2x of what the orbital period would be at the surface.

There is for example [Nyamad C 1](https://www.edsm.net/system/bodies/id/1422/name/Nyamad/details/idB/38597/nameB/Nyamad+C+1) (rotational period was apparently about 1.25x of what the orbital period would be at the surface until it looks like Frontier decided that was too fast and patched it), or [IC 2391 Sector EC-U b3-5 D 1](https://www.edsm.net/system/bodies/id/6831292/name/IC+2391+Sector+EC-U+b3-5/details/idB/7489495/nameB/IC+2391+Sector+EC-U+b3-5+D+1) (rotational period was apparently about 1.08x of what the orbital period would be at the surface until it looks like Frontier decided that was too fast and patched it).

Note also that surface gravity is not set for stars - it would need to be calculated (`StellarMass * 333000 / (Radius / 6378000)^3`)

>>>  Does the planet have signals? And if so, which and how many (bio / geo)?
>>
>> i.e. from the SAASignalsFound event when you map a body? I don't think EDSM currently exposes the counts in its bodies API from where commanders have mapped bodies.
>
> I don't know exactly where the data for counting the signals comes from. The game seems to generate them when scanning the system. It looks like bio signals are "safe" and geo signals are "likely"

Signal counts are only available to the program via the `SAASignalsFound` event, which is written to the player journal by the game when you map the body using the Detailed Surface Scanner.

**Comment by EmergencyPeanut** [comment]
Could I add a request to this for an indication if I have surface scanned a body so they stand out when edsm is on

**Comment by klightspeed** [comment]
So visually distinguish bodies from EDSM vs with local scans, and distinguish bodies with or without a `SAASignalsFound` event?

**Comment by EmergencyPeanut** [comment]
I think the SAASignalsFound would be enough as anything just coming from EDSM wouldn't have that log entry if I understand the network log correctly.

**Comment by robbyxp1** [comment]
I think the consensus of the developers is that adding too much text to the scan graphical display will negate the point of it. If you want detailed info on the object, just like the frontier scan display, you click on it.  The visual text against the planet should be kept to a minimum.  I'll look to see if I can add anything without spoiling the point of the panel.

Maybe we should improve the surveyor panel (which is text based) instead with more data display options.

**Comment by robbyxp1** [comment]
The surveyor, which is text based, will have more info which addresses some of this.

**Comment by kochmich** [comment]
You are certainly right with your considerations. But my request was not aimed at a text representation. My humble wish was to visualize astronomical data in the system view. If this could be represented with options it would be great.

**Comment by robbyxp1** [comment]
I'll keep this in mind, when we next touch the scan panel, to add more display options. For now closing

**Comment by robbyxp1** [comment]
See next version

**Comment by kochmich** [comment]
thank you very much  :-)
